lib/rspec/bash/matchers/called_with_arguments.rb

```diff
@@ -1,14 +1,39 @@
 require 'rspec/expectations'
+require 'differ'
 
-RSpec::Matchers.define :be_called_with_arguments do |*expected_argument_list|
-  chain :times do |expected_invocations|
-    @expected_invocations = expected_invocations
+module CustomMatchers
+  class CalledWithArgumentsMatcher
+    def initialize(*expected_argument_list)
+      @expected_argument_list = expected_argument_list
+    end
+
+    def matches?(actual_command)
+      @actual_command = actual_command
+
+      called_with_correct_args = actual_command.called_with_args?(*@expected_argument_list)
+      called_correct_number_of_times = @expected_invocations ? actual_command.call_count(*@expected_argument_list) == @expected_invocations : true
+
+      called_with_correct_args && called_correct_number_of_times
+    end
+
+    def times(expected_invocations)
+      @expected_invocations = expected_invocations
+      self
+    end
+    
+    def failure_message
+      actual = [@actual_command.call_log.get_call_log_args.first.join(', ').gsub(/\,/,"")].first
```

Comment:
This is no good. If the command is called multiple times in a script (multiple instances of get_call_log_args), then this gets really weird. I'm not sure how to solve this though, so we'll have to figure that out